File: frontend/lib/core/field_limits.dart
abstract final class FieldLimits {
  static const username = 80;
  static const email = 120;
  static const fullName = 120;
  static const roleName = 80;
  static const roleDescription = 255;
  static const invoiceNotes = 500;
  static const paymentNotes = 300;
  static const paymentMethod = 40;
  static const promiseNotes = 1000;
  static const emailSubject = 500;
  static const emailBody = 20000;

  static const dueDateHorizonDays = 365;

  static const documentDescription = 500;

  static const documentFilename = 260;

  static const documentMaxBytes = 10485760;

  /// The server's FieldLimits.TASK_TITLE / TASK_NOTES. Both are plain varchar columns and a save
  /// longer than these is a 400, so the boxes stop at the same number rather than letting somebody
  /// type a paragraph the server will refuse (A6).
  static const taskTitle = 200;
  static const taskNotes = 2000;

  static const gmailClientId = 300;
  static const gmailClientSecret = 300;
  static const gmailRefreshToken = 2000;
}

File: backend/src/main/java/com/geneinvoice/common/FieldLimits.java
package com.geneinvoice.common;

public final class FieldLimits {

    private FieldLimits() {}

    public static final int USERNAME = 80;
    public static final int EMAIL = 120;
    public static final int FULL_NAME = 120;
    public static final int ROLE_NAME = 80;
    public static final int ROLE_DESCRIPTION = 255;
    public static final int PHONE = 30;
    public static final int ADDRESS = 500;
    public static final int PRODUCT_NAME = 150;
    public static final int PRODUCT_DESCRIPTION = 500;
    public static final int INVOICE_NOTES = 500;
    public static final int PAYMENT_NOTES = 300;
    public static final int PAYMENT_METHOD = 40;
    public static final int PROMISE_NOTES = 1000;
    public static final int REASON = 500;
    public static final int DISPUTE_TEXT = 2000;
    public static final int EMAIL_SUBJECT = 500;
    public static final int EMAIL_BODY = 20000;
    public static final int DOCUMENT_FILENAME = 260;
    public static final int DOCUMENT_DESCRIPTION = 500;
    public static final int GMAIL_CLIENT_ID = 300;
    public static final int GMAIL_CLIENT_SECRET = 300;
    public static final int GMAIL_REFRESH_TOKEN = 2000;

    // A task title is a line somebody reads in a list, and the notes are the paragraph
    // behind it; both are varchar and never @Lob, which on Postgres would store an OID
    // instead of readable text (A6).
    public static final int TASK_TITLE = 200;
    public static final int TASK_NOTES = 2000;

    // A rule's own name and the sentence under it. Both are plain varchar; the TEMPLATE fields a
    // rule carries (a task title, an email subject, an email body) are bounded at save time by
    // TASK_TITLE, TASK_NOTES, EMAIL_SUBJECT and EMAIL_BODY above, because a template and the text
    // it renders to land in the same columns and must not have two different ceilings (A1, A3).
    public static final int RULE_NAME = 200;
    public static final int RULE_DESCRIPTION = 1000;
}
